finality-aleph/src/sync_oracle.rs

```diff
@@ -17,14 +20,19 @@ const MAJOR_SYNC_THRESHOLD: Duration = Duration::from_secs(10);
 pub struct SyncOracle {
     last_far_behind: Arc<Mutex<Instant>>,
     last_update: Arc<Mutex<Instant>>,
+    // TODO: remove when SyncingService is no longer needed
+    is_major_syncing: Arc<AtomicBool>,
 }
 
 impl SyncOracle {
-    pub fn new() -> Self {
-        SyncOracle {
+    pub fn new() -> (Self, Arc<AtomicBool>) {
+        let is_major_syncing = Arc::new(AtomicBool::new(false));
+        let oracle = SyncOracle {
             last_update: Arc::new(Mutex::new(Instant::now() - OFFLINE_THRESHOLD)),
             last_far_behind: Arc::new(Mutex::new(Instant::now())),
-        }
+            is_major_syncing: is_major_syncing.clone(),
+        };
+        (oracle, is_major_syncing)
```

Comment:
Not confident about it so I'll put it as a Question:
Don't we need to call `self.major_sync()` here as well?

Reply:
oh, you mean I should initialize the atomic to `true`? hmm, good question, @timorleph do we really want to start in the "major sync" state? this is how it currently works, since we set `last_far_behind` to `now`, right? I guess it makes sense IRL, but in e2e tests we need to wait 10 seconds in the beginning

Reply:
Hmm, e2e tests shouldn't be a problem, as they should start working as soon as you communicate with neighbours. Setting it to `true` should at most impact transactions then, which... meh, shouldn't be that much of a problem.

Reply:
@woocash2 I've changed the initial value to `true`

Reply:
Aight, whatever you think is appropriate